docs/enhancements/metricsConfig_redesign.md

```diff
@@ -0,0 +1,166 @@
+<table>
+<tr>
+<th>Title</th>
+<th>Author</th>
+<th colspan="7">Reviewers</th>
+<th>Approver</th>
+</tr>
+<tr>
+<td>HiveConfig.spec.metricsConfig redesign</td>
+<td>@suhanime</td>
+<td>@bmeng</td>
+<td>@dustman9000</td>
+<td>@tzvatot</td>
+<td>@patjlm</td>
+<td>@janboll</td>
+<td>@berenss</td>
+<td>@hongkailiu</td>
+<td>@2uasimojo</td>
+</tr>
+</table>
+
+# HiveConfig.spec.metricsConfig redesign
+
+[HIVE-2344](https://issues.redhat.com/browse/HIVE-2344)
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Current implementations](#current-implementations)
+    - [Duration based threshold](#duration-based-threshold)
+    - [Additional label support](#additional-label-support)
+  - [Goals](#goals)
+  - [Non Goals](#non-goals)
+- [Proposal](#proposal)
+  - [metricsToReport](#metricstoreport)
+    - [metricNames](#metricnames)
+    - [minimumDuration](#minimumduration)
+    - [additionalClusterDeploymentLabels](#additionalclusterdeploymentlabels)
+    - [clusterDeploymentLabelSelector](#clusterdeploymentlabelselector)
+  - [Implementation Details / Notes](#implementation-details--notes)
+    - [Deprecation](#deprecation)
+    - [Failure Modes](#failure-modes) 
+  - [Risks and Mitigations](#risks-and-mitigations)
+
+## Summary
+As a Hive user interested in customizing metrics, I would like to specify customizations for individual metrics.
```

Comment:
Problem statement is not entirely clear to me. It seems like there a multiple issues simmering regarding "how to effectively operate hive" and they are all trying to be solved with how hive produces metrics. 

I think for an effective metrics solution, we need a concrete operational narrative that defines the role of metrics within it. Currently metrics are being proposed as solutions to:
- observability
- alerting
- troubleshooting (inferring this)

IMO, stake holder expectations need to defined and agreed upon; including those that are not RH SRE. Then we can assess the solution against the problem statement.

Reply:
I changed the summary to be more precise - does it help?
I can see your point about having concrete operational narrative around metrics, but to be clear, hive's job is to publish the metrics, and it is upto the consumer how they want to use them. You can very well use the same metric for both observability and alerting. I don't think prometheus and metrics are the solutions for troubleshooting, though you can definitely use them to notice the patterns.
I would also want to point out - as a part of this enhancement, we're just changing some internal code to make things easier - we're not introducing any new metrics. We've also gotten it reviewed from all major consumers of hive.